packages/liveblocks-node/src/client.ts

```diff
@@ -101,7 +101,7 @@ export type CreateThreadOptions<M extends BaseMetadata> = {
   roomId: string;
   data: {
     comment: { userId: string; createdAt?: Date; body: CommentBody };
-  } & (Record<string, never> extends M ? { metadata?: M } : { metadata: M });
+  } & PartialUnless<M, { metadata: M }>;
```

Comment:
@marcbouchenoire The central way I found here to DRY up these definitions, is using this `PartialUnless<X, T>` helper, which will be `Partial<{ metadata: M }>` (= `{ metadata?: M }`) if the empty object can be assigned to type `X`, but `{ metadata: M }` otherwise.